File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        while True:
            a=str(n)
            p=1
            for digit in str(n):
                p *= int(digit)
            if p%t==0:
                return n
            n+=1
        return n

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int product(int n){
        int product1=1;
        while(n>0){
            int x=n%10;
            product1=product1*x;
            n=n/10; 
        }
        return product1;
    }
    int smallestNumber(int n, int t) {
        int y=n;
        
        while(true){
            if(product(y)%t==0)
            return y;
    
            y++;
        
        
        }
        return y;
        
    }
};